src/modality-agnostic-files/provenance.md

```diff
@@ -0,0 +1,771 @@
+
+```JSON
+{
+    "description": {
+        "Description": "Description of the provenance file(s)."
+    }
+}
+```
+
+## Consistency and uniqueness of identifiers
+
+The following rules and conventions are provided in order to have consistent, human readable, and explicit [IRIs](https://www.w3.org/TR/json-ld11/#iris) as identifiers for JSON objects related to provenance.
+
+!!! note
+    The `Id` field contains the identifier of a JSON object related to provenance.
+
+### Identifiers for provEntities
+
+The identifier of a provEntity describing a BIDS file available in the dataset MUST be a [BIDS URI](../common-principles.md#bids-uri).
+The identifier of a provEntity describing a BIDS file not available in the dataset SHOULD be a [BIDS URI](../common-principles.md#bids-uri) with a fragment part.
+
+!!! warning
+    The use of BIDS URIs may require to define the `DatasetLinks` object in [`dataset_description.json`](dataset-description.md#dataset_descriptionjson).
+
+In other cases, the identifier of a provEntity described in a BIDS dataset `<dataset-name>` MAY have the following form, where `<label>` is an arbitrary value for identifying the provEntity.
+
+```text
+bids:[<dataset-name>]:prov#provEntity-<label>
+```
+
+!!! example "Examples of identifiers for provEntities"
+    - `bids:ds001734:sub-002/anat/sub-02_T1w.nii` - a provEntity describing a T1w file for subject `sub-002` in the `ds001734` dataset;
+    - `bids::sub-014/func/sub-014_task-MGT_run-01_events.tsv` - a provEntity describing an events file for subject `sub-014` in the current dataset;
+    - `bids:fmriprep:sub-001/func/sub-001_task-MGT_run-01_bold_space-MNI152NLin2009cAsym_preproc.nii.gz` - a provEntity describing a bold file for subject `sub-001` in the `fmriprep` dataset;
+    - `bids::prov#provEntity-acea8093` - a provEntity describing a file that is not available in the dataset.
+
+### Identifiers for other objects
+
+The identifier of an activity, software, or environment described in a BIDS dataset `<dataset-name>` SHOULD have the following form, where `<label>` is a human readable name for coherently identifying the object and `<uid>` is a unique group of chars.
+
+```text
+bids:[<dataset-name>]:prov#<label>-<uid>
+```
+
+The uniqueness of this identifier MUST be used to distinguish any activity, software, or environment that are different in any of their attributes.
+
+!!! example "Examples of identifiers for activities, environments and software"
+    - `bids:ds001734:prov#conversion-xfMMbHK1` - a conversion activity described inside the `ds001734` dataset;
+    - `bids::prov#fedora-uldfv058` - a Fedora based environment described inside the current dataset.
+    - `bids:preprocessing:prov#fmriprep-r4kzzMt8` - the fMRIPrep software described inside the `preprocessing` dataset.
+
+## Provenance from an RDF perspective
+
+!!! note
+    The [Resource Description Framework (RDF)](https://www.w3.org/RDF/) is a method to describe and exchange graph data.
```

Comment:
I think we could remove this note and instead add a parenthesis after "RDF graph" below. See proposed edit